managed-inference/images/llama-cpp/request-guard/main_test.go

```
@@ -0,0 +1,668 @@
+	}
+}
+
+func TestGuardPropagatesCancellation(t *testing.T) {
+	started := make(chan struct{})
+	cancelled := make(chan struct{})
+	release := make(chan struct{})
+	var releaseOnce sync.Once
+	releaseBackend := func() { releaseOnce.Do(func() { close(release) }) }
+	backend := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, request *http.Request) {
+		_, _ = io.Copy(io.Discard, request.Body)
+		request.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `request.Body.Close` is not checked

(errcheck)